src/frameworks/compose/discover/runtime/node.ts

```diff
@@ -0,0 +1,276 @@
+    }
+
+    return command;
+  }
+
+  getBuildCommand(
+    packageManager: string,
+    scripts: Record<string, string> | undefined,
+    matchedFramework: FrameworkSpec | null
+  ): Command {
+    let buildCommand: Command = { cmd: "" };
+    if (scripts?.build) {
+      buildCommand.cmd = this.executeScript(packageManager, "build");
+    } else if (matchedFramework && matchedFramework.commands?.build) {
+      buildCommand = matchedFramework.commands.build;
+      buildCommand = this.executeFrameworkCommand(packageManager, buildCommand);
+    }
+
+    return buildCommand;
+  }
+
+  getDevCommand(
+    packageManager: string,
+    scripts: Record<string, string> | undefined,
+    matchedFramework: FrameworkSpec | null
+  ): Command {
+    let devCommand: Command = { cmd: "", env: { NODE_ENV: "dev" } };
+    if (scripts?.dev) {
+      devCommand.cmd = this.executeScript(packageManager, "dev");
+    } else if (matchedFramework && matchedFramework.commands?.dev) {
+      devCommand = matchedFramework.commands.dev;
+      devCommand = this.executeFrameworkCommand(packageManager, devCommand);
+    }
+
+    return devCommand;
+  }
+
+  getRunCommand(
+    packageManager: string,
+    scripts: Record<string, string> | undefined,
+    matchedFramework: FrameworkSpec | null
+  ): Command {
+    let runCommand: Command = { cmd: "", env: { NODE_ENV: "production" } };
+    if (scripts?.start) {
+      runCommand.cmd = this.executeScript(packageManager, "start");
+    } else if (matchedFramework && matchedFramework.commands?.run) {
+      runCommand = matchedFramework.commands.run;
+      runCommand = this.executeFrameworkCommand(packageManager, runCommand);
+    }
+
+    return runCommand;
+  }
+
+  async analyseCodebase(
+    fs: FileSystem,
+    allFrameworkSpecs: FrameworkSpec[]
+  ): Promise<RuntimeSpec | null> {
+    try {
+      const packageJSONRaw = await readOrNull(fs, PACKAGE_JSON);
+      if (!packageJSONRaw) {
+        return null;
+      }
+      const packageJSON = JSON.parse(packageJSONRaw) as PackageJSON;
+      const packageManager = await this.getPackageManager(fs);
+      const nodeImage = this.getNodeImage(packageJSON.engines);
+      const dependencies = await this.getDependencies(fs, packageJSON, packageManager);
+      const matchedFramework = await frameworkMatcher(
+        NODE_RUNTIME_ID,
+        fs,
+        allFrameworkSpecs,
+        dependencies
+      );
+
+      const runtimeSpec: RuntimeSpec = {
+        id: NODE_RUNTIME_ID,
+        baseImage: nodeImage,
+        packageManagerInstallCommand: this.packageManagerInstallCommand(packageManager),
+        installCommand: this.installCommand(packageManager),
+        detectedCommands: this.detectedCommands(
+          packageManager,
+          packageJSON.scripts,
+          matchedFramework
+        ),
+      };
+
+      return runtimeSpec;
+    } catch (error: any) {
+      throw new FirebaseError(`Failed to indentify commands for codebase: ${error}`);
```

Comment:
nit*: This could also catch errors like "failed to parse engine"  and the "Failed to identity command" seems less relevant..